### PR TITLE
chore: deprecate TimezoneGMT class

### LINKED ISSUE #2301: Deprecate TimezoneGMT class
There is no need for it in py 3.8+.

### PR DESCRIPTION
**Pull request by CaselIT**
Fixes #2301

### CODE REVIEW

File: docs/_newsfragments/2301.newandimproved.rst
Comment (vytas7) [inline docs/_newsfragments/2301.newandimproved.rst:1]: Release notes should not be worded in imperative as docstrings, just tell a story what has been done.

Moreover, I think deprecations better suit `.misc`, not `.newandimproved`.

File: falcon/util/time.py [hunk 1]
```diff
@@ -20,6 +23,13 @@ class TimezoneGMT(datetime.tzinfo):
 
     GMT_ZERO = datetime.timedelta(hours=0)
 
+    def __init__(self) -> None:
```
Comment (vytas7) [inline falcon/util/time.py:26]: Maybe we could just plop the `@deprecation.deprecated(...)` decorator on `__init__`, just like we do in `falcon.app.API`?

File: tests/test_cookies.py
Comment (vytas7) [inline tests/test_cookies.py:1]: Not critical, but we have tests for various `util` functions, maybe it testing `TimezoneGMT` would better fit there than inside cookies? It is not really related to cookies.

**Review by vytas7 — CHANGES_REQUESTED** [review]

File: tests/test_cookies.py
Comment (CaselIT) [reply tests/test_cookies.py:1]: it was there because it's only used there I think. will move them there

**Review by vytas7 — APPROVED** [review]
Thanks, it makes sense to deprecate this class indeed as it brings not much value over simply using UTC. :+1:

As discussed on Gitter, technically GMT and UTC are **not** the same, but CPython does not really support, itself, the subtle differences between the two time systems. (See also: https://github.com/python/cpython/issues/67762,)

**Comment by CaselIT** [comment]
I don't actually think we need gmt in the cookies. It's very likely that if the specification ever mantated gmt it has now amended to mandata utc, since that's the one that's standard.

**Pull request merged**